ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/StartupUtil.java

```diff
@@ -158,4 +159,40 @@ public static Store get_genesis_store(BeaconStateWithCache genesis_state) {
         block_states,
         checkpoint_states);
   }
+
+  public static ChainStorageClient initChainStorageClient(
+      final EventBus eventBus,
+      final long genesisTime,
+      final String startState,
+      final int numValidators) {
+    final ChainStorageClient chainStorageClient =
+        ChainStorage.Create(ChainStorageClient.class, eventBus);
+    initChainStorageClient(chainStorageClient, genesisTime, startState, numValidators);
+    return chainStorageClient;
+  }
+
+  public static void initChainStorageClient(
+      final ChainStorageClient chainStorageClient,
+      final long genesisTime,
+      final String startState,
+      final int numValidators) {
+    chainStorageClient.setGenesisTime(UnsignedLong.valueOf(genesisTime));
+    BeaconStateWithCache initialState;
+    if (startState != null) {
```

Comment:
(optional) Might be worth making this method private and splitting the public methods into two: `initChainStorageClient()` and `initMockChainStorageClient ` or something

Reply:
I can see what you're getting at, but can't quite find the right way to make things fit really right. I also have a nagging suspicion that this code is going to create issues when we're not using a mocked start as it will load a genesis state before the ETH1 chain events have actually triggered the real genesis.

And the interop issue with lighthouse suggests we're getting the wrong value out of our startup somehow anyway so I'm going to be back in this kind of area soon I think.